rules_building_block/discovery_generic_registry_query.toml

```diff
@@ -36,8 +37,8 @@ query = '''
 host.os.type:windows and event.category:process and event.type:start and (
   (process.name.caseless:"reg.exe" and process.args:"query") or 
   (process.name.caseless:("powershell.exe" or "powershell_ise.exe" or "pwsh.exe") and 
-   process.command_line.caseless:((*Get-ChildItem* or *Get-Item* or *Get-ItemProperty*) and 
-   (*HKCU* or *HKEY_CURRENT_USER* or *HKEY_LOCAL_MACHINE* or *HKLM* or *Registry\:\:*))))
+   process.command_line.caseless:((*get-childItem* or *get-item* or *get-itemproperty*) and
+   (*hkcu* or *hkey_current_user* or *hkey_local_machine* or *hklm* or *registry\:\:*))))
```

Comment:
Is this less expensive?

Reply:
that was not about performance so much as fixing a bug (`.caseless`)

The way to fix performance here would be

>However, as mentioned above, we should consider refining this further with process.args to eliminate the wildcards. As it is right now, this is very brittle because of the possible occurrence of aliases or shorthand for the cmdlets. A more performant and accurate approach would be to convert them to process.args and iterate the known alternatives, such as

>process.args:((gci or get-childitem or Get-ChildItem or gi or get-item or Get-Item or ...) and
   (hkcu or HKCU or hkey_current_user or HKEY_CURRENT_USER ...)))

Reply:
Performance wide, I wonder if these are parsed in the `process.args` or command line args and thus we do not need wildcards and can do string comparison against the array. This may help significantly